test/unit/org/apache/cassandra/repair/FuzzTest.java

```diff
@@ -0,0 +1,1453 @@
+        CLOCK_GLOBAL.setString(ClockAccess.class.getName());
+        DatabaseDescriptor.daemonInitialization();
+        DatabaseDescriptor.setPartitionerUnsafe(Murmur3Partitioner.instance); // TOOD (coverage): random select
+        IFailureDetector fd = Mockito.mock(IFailureDetector.class);
+        Mockito.when(fd.isAlive(Mockito.any())).thenReturn(true);
+        DatabaseDescriptor.setFailureDetector(() -> fd);
+        StreamingChannel.Factory.Global.unsafeSet(new StreamingChannel.Factory()
+        {
+            private final AtomicInteger counter = new AtomicInteger();
+
+            @Override
+            public StreamingChannel create(InetSocketAddress to, int messagingVersion, StreamingChannel.Kind kind) throws IOException
+            {
+                StreamingChannel mock = Mockito.mock(StreamingChannel.class);
+                int id = counter.incrementAndGet();
+                StreamSession session = Mockito.mock(StreamSession.class);
+                StreamReceiveException access = new StreamReceiveException(session, "mock access rejected");
+                StreamingDataInputPlus input = Mockito.mock(StreamingDataInputPlus.class, invocationOnMock -> {
+                    throw access;
+                });
+                Mockito.doNothing().when(input).close();
+                Mockito.when(mock.in()).thenReturn(input);
+                Mockito.when(mock.id()).thenReturn(id);
+                Mockito.when(mock.peer()).thenReturn(to);
+                Mockito.when(mock.connectedTo()).thenReturn(to);
+                Mockito.when(mock.send(Mockito.any())).thenReturn(ImmediateFuture.success(null));
+                Mockito.when(mock.close()).thenReturn(ImmediateFuture.success(null));
+                return mock;
+            }
+        });
+        ExecutorFactory delegate = ExecutorFactory.Global.executorFactory();
+        ExecutorFactory.Global.unsafeSet(new ExecutorFactory()
+        {
+            @Override
+            public LocalAwareSubFactory localAware()
+            {
+                return delegate.localAware();
+            }
+
+            @Override
+            public ScheduledExecutorPlus scheduled(boolean executeOnShutdown, String name, int priority, SimulatorSemantics simulatorSemantics)
+            {
+                return delegate.scheduled(executeOnShutdown, name, priority, simulatorSemantics);
+            }
+
+            private boolean shouldMock()
+            {
+                return StackWalker.getInstance().walk(frame -> {
+                    StackWalker.StackFrame caller = frame.skip(3).findFirst().get();
+                    if (caller.getClassName().startsWith("org.apache.cassandra.streaming.")) return true;
+                    return false;
+                });
+            }
+
+            @Override
+            public Thread startThread(String name, Runnable runnable, InfiniteLoopExecutor.Daemon daemon)
+            {
+                if (shouldMock()) return new Thread();
```

Comment:
nit: Does it matter if we don't start the new thread?

Reply:
streaming doesn't touch the response so kinda not...